# RNA-seq and Tn-seq reveal fitness determinants of vancomycin-resistant _Enterococcus faecium_ during growth in human serum
This project was done using Zhang _et al.(2017)_ ["RNA-seq and Tn-seq reveal fitness determinants of vancomycin-resistant Enterococcus faecium during growth in human serum"](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-017-4299-9) article. 

The workflow of the project is detailed in the [Project Planning](https://github.com/mariagadeam/GenomeAnalysis/wiki/Project-planning#pipeline-of-analyses-and-softwares) part of the Wiki.
The code was structured in three directories: 
* **01_Assembly**: Includes the pipeline for preprocessing and quality check of the Illumina reads, the pipeline for assembling the Pacbio long reads with Canu, the pipeline for hybrid SPAdes assembly with Nanopore and Illumina reads, and the pipeline to evaluate both assemblies.
* **02_Annotation**: Includes the pipeline for genome annotation, sinteny comparison against the sequence of another _E.faecium_ strain , and a subdirectory for the SNP calling analysis. This includes the pipeline for SNP calling, a pipeline to extract statistics and metrics about the obtained SNPs, an extra pipeline using data from a similar study to study further SNPs, and a R file to help visualize results from this extra analysis.
* **03_Transcriptome_analyses**: Includes the pipeline for preprocessing of the Illumina RNA-seq data, a subdirectory for the whole mapping, sorting, indexing and counting process, and the pipeline for the differential expression analysis. The mapping subdirectory includes the pipeline for mapping, indexing and sorting of the genome and reads, the pipeline for counting the reads, a pipeline to analyze the percentage of reads that mapped to the genome assembly , and a pipeline to analyze the coverage of each sample.
